runtime-decompiler/src/main/java/org/fife/ui/hex/swing/HexSearch.java

```diff
@@ -0,0 +1,161 @@
+package org.fife.ui.hex.swing;
+
+import java.util.ArrayList;
+
+public class HexSearch {
+
+    private HexEditor hex;
+    private int start = 0;
+    int end = 0;
+    int counter = 0;
+    boolean found = false;
+
+    public enum HexSearchOptions {
+        HEX,
+        INT,
+        TEXT
+    }
+
+    public HexSearch(HexEditor hex) {
+        this.hex = hex;
+    }
+
+    private ArrayList<Byte> getByteArray(String str, HexSearchOptions type) {
+        ArrayList<Byte> arr = new ArrayList<>();
+        switch (type) {
+            case TEXT:
+                byte[] bytesText = str.getBytes();
+                if (bytesText.length == 0) {
+                    throw new StringIndexOutOfBoundsException();
+                }
+                for (byte b : bytesText) {
+                    arr.add(b);
+                }
+                break;
+
+            case INT:
+                if (str.trim().equals("")) {
+                    throw new StringIndexOutOfBoundsException();
+                }
+                String[] spliced = str.split(" ");
+                for (String s : spliced) {
+                    try {
+                        arr.add(Integer.valueOf(s).byteValue());
+                    } catch (NumberFormatException e) {
+                        throw e;
```

Comment:
catch e -> throw e???

Reply:
The exception needs to bubble up so it can be handled by the function calling this function.

Reply:
The exception needs to bubble up so it can be handled by the function calling this function.